### PR TITLE
Add Transfer logic for AutoAccountCreation

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
1. Add tracking to `autoCreatedAccount` in `SideEffectsTracker`
2. Add `alias` to `BalanceChange`
3. Add validation for `alias` to be a `primitiveKey` in `PureTransferSemanticChecks`
4. Modify `TransferLogic` to auto create accounts when the transfer has only `alias` set on AccountId, which is done in `AutoAccountCreateLogic`
5. While saveing ExpiringRecord , add tracking autoCreatedAccount to receipt
6. Add `cryptoCreate` transaction to `SyntheticTxnFactory`
7. Unit tests
8. TODO : EETs

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/ledger/BalanceChange.java [hunk 1]
```diff
@@ -233,4 +244,11 @@ public void setExemptFromCustomFees(boolean exemptFromCustomFees) {
 	public boolean isExemptFromCustomFees() {
 		return exemptFromCustomFees;
 	}
+
+	public boolean hasOnlyAlias() {
+		AutoAccountsManager autoAccounts = AutoAccountsManager.getInstance();
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/ledger/BalanceChange.java:249]: Using a static method for object lookup is an anti-pattern; imo we can:

1. Let Dagger2 create the `AutoAccountsManager` as a singleton in the normal way.
2. Inject that singleton into the `TransferLogic`.
3. Pass it into this `BalanceChange.hasOnlyAlias()` method.

File: hedera-node/src/main/java/com/hedera/services/ledger/TransferLogic.java [hunk 2]
```diff
@@ -49,78 +53,109 @@
...
-            } else {
-                validity = tokenStore.tryTokenChange(change);
-            }
-            if (validity != OK) {
-                break;
-            }
-        }
+	public void transfer(final List<BalanceChange> changes) {
+		var validity = OK;
+		List<ByteString> autoCreationAliases = new ArrayList<>();
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/ledger/TransferLogic.java:90]: We should wait to create this `ArrayList` until we know an auto-creation is being attempted; creating an `ArrayList` is actually fairly heavy, so we don't want to do it unnecessarily 10,000 times a second. 😁

Comment (Neeharika-Sompalli) [reply hedera-node/src/main/java/com/hedera/services/ledger/TransferLogic.java]: Ah yes . Modified to create new list only if accounts to be auto created 😛

File: hedera-node/src/main/java/com/hedera/services/utils/EntityNum.java [hunk 3]
```diff
@@ -69,7 +70,12 @@ public static EntityNum fromAccountId(AccountID grpc) {
 		if (!areValidNums(grpc.getShardNum(), grpc.getRealmNum())) {
 			return MISSING_NUM;
 		}
-		return fromLong(grpc.getAccountNum());
+		final var autoAccountsMap = AutoAccountsManager.getInstance().getAutoAccountsMap();
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hedera-node/src/main/java/com/hedera/services/utils/EntityNum.java]: yes , removed this change.

File: hedera-node/src/main/java/com/hedera/services/txns/crypto/AutoAccountCreateLogic.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	 */
+	public ResponseCodeEnum createAutoAccounts(final List<ByteString> accountsToBeCreated,
+			final TransactionalLedger<AccountID, AccountProperty, MerkleAccount> accountsLedger) {
+		final var sideEffects = sideEffectsFactory.get();
+		for (ByteString alias : accountsToBeCreated) {
+			try {
+				/* create a crypto create synthetic transaction */
+				var syntheticCreateTxn = syntheticTxnFactory.cryptoCreate(alias, 0L);
+				var newAccountId = ids.newAccountId(syntheticCreateTxn.getTransactionID().getAccountID());
+				accountsLedger.create(newAccountId);
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hedera-node/src/main/java/com/hedera/services/txns/crypto/AutoAccountCreateLogic.java:98]: Changed . Thanks !

**Pull request merged**